util/timers/monotonic.go

```diff
@@ -23,71 +23,81 @@ import (
 	"github.com/algorand/go-algorand/protocol"
 )
 
+type timeout struct {
+	delta time.Duration
+	ch    <-chan time.Time
+}
+
 // Monotonic uses the system's monotonic clock to emit timeouts.
-type Monotonic struct {
+type Monotonic[TimeoutType comparable] struct {
 	zero     time.Time
-	timeouts map[time.Duration]<-chan time.Time
+	timeouts map[TimeoutType]timeout
```

Comment:
I do not understand how's the old code (any duration associated with a channel) is equivalent to the new code (only 3 possible durations determined by TimeoutType). Before `TimeoutAt(delta)` supported invocations with different `delta` values, and now it overwrites `timeout` value for particular `TimeoutType` on every new `delta` argument, making impossible to have two subsequent calls `TimeoutAt(1)`, `TimeoutAt(1)` to return "the same" channel.

Reply:
There is a behavior change here. We noticed that once agreement updates a deadline (for a certain type), it no longer cares about older deadlines (of the same type).

Two subsequent calls to TimeoutAt(1, timeouttype) of a deadline of timeouttype would return the same channel. This behavior should be in line with the existing code.

Reply:
right about subsequent, I meant non-subsequent. Thanks for explaining.